src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,28 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  var foundKey;
+  for (let key of Object.keys(config.carry.sizes).sort(function(a, b) { return parseInt(a, 10) - parseInt(b, 10); })) {
```

Comment:
properties in JS objects are sorted according to "original insertion order". In our case of object literal, order would be the same as in our literal